### PR TITLE
Make UI tests retry if error nodes show up, try to harden file browse…

### PR DESCRIPTION
**Pull request by abrooksv**
…r more

* If we load an error node in the explorer, wait a few seconds and hit retry
* Change how we enter text into the file browser due to focus kept getting stolen

File browser logs:
```
                                 Set file path to /var/folders/q5/cfv4k63519vcllnnpwz5zsj0wnjvck/T/junit14382372376297587214
13:37:23.789 INFO  -                 Search 'JTextFieldFixture' by '//div[@class='BorderlessTextField']'
13:37:23.866 INFO  -                 Set text ''
13:37:23.919 INFO  -                 Type path '/var/folders/q5/cfv4k63519vcllnnpwz5zsj0wnjvck/T/junit14382372376297587214'
13:37:23.919 INFO  -                     Set text '/'
13:37:24.063 INFO  -                     Set text '/var'
13:37:24.213 INFO  -                     Set text '/var/folders'
13:37:24.363 INFO  -                     Set text '/var/folders/q5'
13:37:24.513 INFO  -                     Set text '/var/folders/q5/cfv4k63519vcllnnpwz5zsj0wnjvck'
13:37:24.666 INFO  -                     Set text '/var/folders/q5/cfv4k63519vcllnnpwz5zsj0wnjvck/T'
13:37:24.817 INFO  -                     Set text '/var/folders/q5/cfv4k63519vcllnnpwz5zsj0wnjvck/T/junit14382372376297587214'
13:37:24.968 INFO  -             Search 'ComponentFixture' by '//div[@accessiblename='Refresh']'
13:37:25.109 INFO  -             ..click
13:37:25.814 INFO  -             requireSelection /|var|folders|q5|cfv4k63519vcllnnpwz5zsj0wnjvck|T|junit14382372376297587214
```

Explorer node logs:
```
    13:38:13.337 INFO  -     Open editor for bucket uitest-20201222-ebe16daf-17f2-47c4-afd9-067278437643
    13:38:13.337 INFO  -         Search 'JTreeFixture' by '//div[@class='Tree']'
    13:38:13.355 INFO  -         expandPath S3
    13:38:18.525 INFO  -         Error node was returned, will wait and try again in PT2S . Attempt 1 of 5
    13:38:20.531 INFO  -             Pressing Refresh...
    13:38:20.531 INFO  -                 Search 'ComponentFixture' by '//div[@accessiblename='Refresh AWS Connection']'
    13:38:20.550 INFO  -                 ..click
    13:38:21.265 INFO  -         expandPath S3
```

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/AwsExplorer.kt [hunk 1]
```diff
@@ -5,44 +5,77 @@ package software.aws.toolkits.jetbrains.uitests.fixtures
...
+import org.assertj.swing.timing.Pause
 import java.time.Duration
 
 fun RemoteRobot.awsExplorer(
     timeout: Duration = Duration.ofSeconds(20),
     function: AwsExplorer.() -> Unit
 ) {
     step("AWS explorer") {
-        find<AwsExplorer>(byXpath("//div[@class='ExplorerToolWindow']"), timeout).apply(function)
+        find<AwsExplorer>(byXpath("//div[@accessiblename='AWS Explorer Tool Window' and @class='InternalDecorator']"), timeout).apply(function)
```
Comment (abrooksv) [inline ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/AwsExplorer.kt:20]: This changed due to it needs to contain both the content (ExplorerToolWindow) and the action stripe so we can find our action buttons

File: ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/AwsExplorer.kt [hunk 2]
```diff
@@ -5,44 +5,94 @@ package software.aws.toolkits.jetbrains.uitests.fixtures
...
+    private fun expandServiceNode(serviceName: String) {
+        step("Expand service node '$serviceName'") {
+            for (attempt in 1..MAX_ATTEMPTS) {
+                val explorerTree = explorerTree()
+
+                waitFor { explorerTree.hasPath(serviceName) }
+                explorerTree.expandPath(serviceName)
+                explorerTree.waitUntilLoaded()
+
+                step("DEBUG: ${explorerTree.findAllText().joinToString { it.text }}") {}
```
Comment (rli) [inline ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/AwsExplorer.kt:66]: is there a debug logger?

File: ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/JTreeFixture.kt [hunk 3]
```diff
@@ -61,3 +69,17 @@ class JTreeFixture(
         }
     }
 }
+
+fun JTreeFixture.waitUntilLoaded() {
+    step("waiting for loading text to go away...") {
+        try {
+            while (true) {
+                Pause.pause(100)
+                findAllText().any { it.text.equals("loading...", ignoreCase = true) }
```
Comment (rli) [inline ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/JTreeFixture.kt:78]: IIRC we had some trees that override the default loading text

Comment (hunterwerlla) [reply ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/JTreeFixture.kt:78]: We do, this should either be overridable probably

File: ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/AwsExplorer.kt (hunk 2 above)
Comment (abrooksv) [reply ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/AwsExplorer.kt:66]: I'll delete it when I am done with this PR.

File: ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/JTreeFixture.kt (hunk 3 above)
Comment (abrooksv) [reply ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/JTreeFixture.kt:78]: Which trees?

Comment (hunterwerlla) [reply ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/JTreeFixture.kt:78]: Seems to only be tables for now, so we can think about this later if we need to

**Comment by abrooksv** [comment]
After running all these tests multiple times throughout this PR, the original change to try to retry on throttling with backoff has not improved anything so removing that logic to simplify it

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**